Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2023 André Hernandez-Espiet. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: André Hernandez-Espiet
+-/
+import Mathlib.Data.Real.Basic
+import Mathlib.Data.Set.Finite
+
+/-!
+Axioms of synthetic geometry
+-/
+
+/-! Universes for points lines and circles-/
+universe u
+
+/-- `IncidenceGeometry` represents geometry in the Euclidean sense, with primitives for points
+lines and circles-/
+class IncidenceGeometry where
+/--Points in the plane-/
+Point : Type u
+/--Lines in the plane-/
+Line : Type u
+/--Circles in the plane-/
+Circle : Type u
```

Comment:
I suspect this typeclass will be more usable as
```suggestion
class IncidenceGeometry (Point : Type u) (Line : outParam <| Type u) (Circle : outParam <| Type u)
```
because then we can add an `instance` that says "euclidean space satisfies the axioms of "incidence geometry".

Reply:
When I try to change this, `variable [i : IncidenceGeometry]` breaks, along with anything that requires points. How do I get around this?

Reply:
You change it to `{Point : Type*} [IncidenceGeometry Point]`

Reply:
Ok. It seems if I do that, then I would have to write `i.Line` or `i.rightangle` every time? Is there some way to avoid this?